### ISSUE TITLE
Conditional tests with Gradle by backward match is failed (5.3.0 or later)

### ISSUE THREAD

**Issue by kohei-akasaka-biz** [kotest/kotest#3121]
**Which version of Kotest are you using**
- Kotest: 5.3.0 or later
- Kotlin: 1.6.10
- Gradle: 7.4.2

Hi, I use version 5.1.0 of kotest. I have updated my version of kotest to 5.3.0.
`./gradlew test --tests '*Spec'` is failed in version 5.3.0 of kotest.
I browsed https://kotest.io/docs/framework/conditional/conditional-tests-with-gradle.html.
The page says I can filter with `gradle test --tests '*IntegrationTest'`, but failed to run on 5.3.0.

```
$./gradlew test --tests '*Spec'
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :test

NumberSpec > NoSuchElementException FAILED
    java.util.NoSuchElementException at _Strings.kt:71

NumberSpec > executionError FAILED
    java.util.NoSuchElementException at _Strings.kt:71
        Caused by: java.util.NoSuchElementException at _Strings.kt:71

2 tests completed, 2 failed

> Task :test FAILED

FAILURE: Build failed with an exception.
```

Is this a bug?

Thanks.

## Status of Different Versions
Assuming you define somepackage.NumberSpec.

||5.1.0|5.3.0|
|:-----: | :-----: | :-----: |
|backward match <br/> (\*Spec)| ✅ | ❌|
|forward match <br/> (somepackage.\*) |✅|✅|

### Files
- build.gradle
```
plugins {
    id 'org.jetbrains.kotlin.jvm' version '1.6.10'
}

group = 'org.example'
version = '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    testImplementation 'io.kotest:kotest-runner-junit5:5.3.0'
}

test {
    useJUnitPlatform()
}

compileTestKotlin {
    kotlinOptions.jvmTarget = '1.8'
}

compileTestKotlin {
    kotlinOptions.jvmTarget = '1.8'
}
```

- Test Target Program File(Number.kt)
```
data class Number(val number: Int) {
    fun addNumber(n: Int): Int {
        return number + n
    }
}
```

- Test Program File (NumberSpec.kt)
```
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class NumberSpec : FunSpec({
    test("test") {
        val number = Number(1)

        val actual = number.addNumber(3)
        actual.shouldBe(4)
    }
})
```

**Comment by sksamuel** [comment]
Can you try 5.4.0 ?
Just in case.

**Comment by kohei-akasaka-biz** [comment]
Thanks for the reply.
Yes, I can.
I ran it, but the result was a failure.

**Comment by sksamuel** [comment]
Doesn't seem to support the *prefix, only suffix* style.
We can look to enhance/fix that for 5.4.2

**Comment by kohei-akasaka-biz** [comment]
Thank you very much.
